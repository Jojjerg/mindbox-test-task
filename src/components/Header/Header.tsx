import React from "react";

const Header = () => {
  return (
    <header data-testid="header">
      <h1
        data-testid="header-text"
        className="font-light text-[64px] text-white"
      >
        todos
      </h1>
    </header>
  );
};

export default Header;
