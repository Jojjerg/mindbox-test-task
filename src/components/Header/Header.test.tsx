import { render, screen } from "@testing-library/react";

import Header from "./Header";
import React from "react";
import { createRenderer } from "react-test-renderer/shallow";

const renderer = createRenderer();

const defaultComponent = <Header />;

describe("<Header />", () => {
  it("should render and match the snapshot", () => {
    renderer.render(defaultComponent);
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });

  it("header-text renders correctly", () => {
    const { getByTestId } = render(defaultComponent);
    expect(getByTestId("header-text")).toBeVisible();
  })
});