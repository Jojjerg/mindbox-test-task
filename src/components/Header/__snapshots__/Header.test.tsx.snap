// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render and match the snapshot 1`] = `
<header
  data-testid="header"
>
  <h1
    className="font-light text-[64px] text-white"
    data-testid="header-text"
  >
    todos
  </h1>
</header>
`;
